### DISCUSSION TITLE
Python id "hiba"

### DISCUSSION THREAD

**Discussion by Fulop95** [kivy/kivy-github-discussions#36] [Q&A] [answered]
Sziasztok! Valamiért nem működik a program. Ha nem írom bele az "id"-t, akkor működik. Valaki tudja esetleg mi a hiba? Van erre valami más kód? Mindenkinek előre is köszönöm a válaszát!

import os
os.environ['KIVY_NO_CONSOLELOG'] = '1'

cwd = os.getcwd()
os.environ['KIVY_HOME'] = cwd + '/kivy'

from kivy.app import App
from kivy.uix.button import Button
from kivy.uix.widget import Widget

class AlapWidget(Widget):
    def __init__(self):
        super().__init__()
        gomb1 = Button(text="Gomb 1", size=(100, 50), pos=(0,0), id="gomb1")
        gomb1.bind(on_press=self.hello_gomb)
        self.add_widget(gomb1)

        gomb2 = Button(text="Gomb 2", size=(100, 50), pos=(100, 100), id="gomb2")
        gomb2.bind(on_press=self.hello_gomb)
        self.add_widget(gomb2)

    def hello_gomb(self, instance):
        if instance.id == "gomb1":
            print("Hello Gomb 1")
        elif instance.id == "gomb2":
            print("Hello Gomb 2")

class TesztApp(App):
    def build(self):
        return AlapWidget()

TesztApp().run()

![Képernyőkép 2024-05-08 143551](https://github.com/kivy/kivy-github-discussions/assets/169263460/e384d518-cca7-48ee-9ad7-025d289bd3eb)

**Comment by obragaa** [comment, marked as answer]
A hiba a programban van, de könnyen megoldható. Az "id" attribútumot hibásan használtad a gombokon. Ha elhagyod az "id" attribútumot, akkor működik a program. A javított kód a következőképpen néz ki:

```python
import os

cwd = os.getcwd()
os.environ['KIVY_HOME'] = cwd + '/kivy'

from kivy.app import App
from kivy.uix.button import Button
from kivy.uix.widget import Widget

class AlapWidget(Widget):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        gomb1 = Button(text="Gomb 1", size=(100, 50), pos=(0,0))
        gomb1.bind(on_press=self.hello_gomb)
        self.add_widget(gomb1)

        gomb2 = Button(text="Gomb 2", size=(100, 50), pos=(100, 100))
        gomb2.bind(on_press=self.hello_gomb)
        self.add_widget(gomb2)

    def hello_gomb(self, instance):
        if instance.text == "Gomb 1":
            print("Hello Gomb 1")
        elif instance.text == "Gomb 2":
            print("Hello Gomb 2")

class TesztApp(App):
    def build(self):
        return AlapWidget()

TesztApp().run()
```

Ezzel a kóddal minden gomb jól fog működni, anélkül, hogy az "id" attribútumot kellene használni. Remélem, ez segít a probléma megoldásában!

**Reply by Fulop95** [reply]
Szia! Köszönöm a segítséget! Így már működik a program
